geopandas/plotting.py

```diff
@@ -535,17 +536,38 @@ class identifier and the bins.
 
     """
     try:
-        from pysal.esda.mapclassify import (
-            Quantiles, Equal_Interval, Fisher_Jenks)
+        from mapclassify import (
+            Quantiles, Equal_Interval, Fisher_Jenks,
+            Fisher_Jenks_Sampled)
         schemes = {}
         schemes['equal_interval'] = Equal_Interval
         schemes['quantiles'] = Quantiles
         schemes['fisher_jenks'] = Fisher_Jenks
+        schemes['fisher_jenks_sampled'] = Fisher_Jenks_Sampled
+
         scheme = scheme.lower()
         if scheme not in schemes:
             raise ValueError("Invalid scheme. Scheme must be in the"
                              " set: %r" % schemes.keys())
         binning = schemes[scheme](values, k)
         return binning
     except ImportError:
-        raise ImportError("PySAL is required to use the 'scheme' keyword")
+        try:
+            from mapclassify.api import (
+                Quantiles, Equal_Interval, Fisher_Jenks,
+                Fisher_Jenks_Sampled)
+            schemes = {}
+            schemes['equal_interval'] = Equal_Interval
+            schemes['quantiles'] = Quantiles
+            schemes['fisher_jenks'] = Fisher_Jenks
+            schemes['fisher_jenks_sampled'] = Fisher_Jenks_Sampled
+
+            scheme = scheme.lower()
+            if scheme not in schemes:
+                raise ValueError("Invalid scheme. Scheme must be in the"
+                                 " set: %r" % schemes.keys())
+            binning = schemes[scheme](values, k)
+            return binning
+        except ImportError:
+            raise ImportError("mapclassify is required to use the 'scheme'"
```

Comment:
Can you put the try/except only around the import statement? Then we don't need to repeat the rest of the code (the construction of the `schemes` and `binning`)